server/src/test/java/io/crate/replication/logical/action/PublicationsStateActionTest.java

```diff
@@ -56,11 +65,21 @@ public void removeLogger() {
 
     @Test
     public void test_resolve_relation_names_for_all_tables_ignores_table_with_soft_delete_disabled() throws Exception {
+        // Mock user having READ_WRITE_DEFINE privileges on the TABLE doc.t1 but not on doc.t3
+        var user = new User("some_user", Set.of(), Set.of(), null) {
+            @Override
+            public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, String ident, String defaultSchema) {
+                return (READ_WRITE_DEFINE.contains(type) && clazz.equals(Privilege.Clazz.TABLE) && ident.equals("doc.t1"));
```

Comment:
This would break the test as `doc.t2` is now filtered out by the user rule.
Please create a dedicated test instead of adding a different case to this test.